driver/drivers.go

```diff
@@ -164,5 +168,23 @@ func (d *Drivers) Delete(taskName string) error {
 	}
 
 	delete(d.drivers, taskName)
+	delete(d.deletion, taskName)
 	return nil
 }
+
+func (d *Drivers) MarkForDeletion(name string) {
+	d.mu.Lock()
+	defer d.mu.Unlock()
+	d.deletion[name] = true
+}
+
+func (d *Drivers) IsMarkedForDeletion(name string) bool {
+	d.mu.Lock()
+	defer d.mu.Unlock()
```

Comment:
Since this method only deals with reading from the map, the mutex could be better leveraged using the read lock so multiple reads can occur at the same time. This way it will only block if there's an active write lock.

```suggestion
	d.mu.RLock()
	defer d.mu.RUnlock()
```